File: bernievshillary/IssuePosition.m
//
//  IssuePosition.m
//  bernievshillary
//
//  Created by emersonmalca on 5/16/16.
//
//

#import "IssuePosition.h"

@interface IssuePosition ()

@property (nonatomic, readwrite) IssuePositionType type;
@property (strong, nonatomic, readwrite, nullable) NSString *text;

@end

@implementation IssuePosition

+ (nonnull IssuePosition *)issuePositionWithType:(IssuePositionType)type {
    return [self issuePositionWithType:type text:nil];
}

+ (nonnull IssuePosition *)issuePositionWithType:(IssuePositionType)type text:(nullable NSString *)text {
    IssuePosition *issue = [[IssuePosition alloc] init];
    issue.type = type;
    issue.text = text;
    return issue;
}

+ (IssuePositionType)typeFromNumber:(NSNumber *)number {
    switch ([number intValue]) {
        case 1:
            return IssuePositionTypeFor;
            break;
        case 0:
            return IssuePositionTypeAgainst;
        default:
            return IssuePositionTypeNeutral;
            break;
    }
}

@end

File: bernievshillary/IssuePosition.swift
//
//  IssuePosition.swift
//  
//
//  Created by Adam Barr-Neuwirth on 8/3/16.
//
//
import Foundation

@objc public class IssuePosition: NSObject {

    var type: IssuePositionType = .Neutral
    var text: NSString?
    
    @objc enum IssuePositionType: NSInteger {
        case For
        case Against
        case Neutral
    }
    
    class func issuePositionWithType(type: IssuePositionType) -> IssuePosition {
        return self.issuePositionWithType(type, text: nil)
    }

    class func issuePositionWithType(type: IssuePositionType, text: NSString?) -> IssuePosition {
        let issue: IssuePosition = IssuePosition()
        issue.type = type
        issue.text = text!
        return issue
    }

    class func typeFromNumber(number: NSNumber) -> IssuePositionType {
        switch CInt(number.intValue) {
            case 1:
                return .For
            case 0:
                return .Against
            default:
                return .Neutral
        }
        
    }
}